app/src/main/java/mozilla/lockbox/presenter/ItemDetailPresenter.kt

```diff
@@ -16,13 +18,32 @@ import mozilla.lockbox.store.DataStore
 interface ItemDetailView {
     var itemId: String?
     fun updateItem(item: ItemDetailViewModel)
+
+    val hostnameClicks: Observable<Unit>
 }
 
 class ItemDetailPresenter(
     private val view: ItemDetailView,
     private val dispatcher: Dispatcher = Dispatcher.shared,
     private val dataStore: DataStore = DataStore.shared
 ) : Presenter() {
+
+    override fun onViewReady() {
+        this.view.hostnameClicks
+                .subscribe {
+                    view.itemId?.let {
```

Comment:
This is the third time I've seen this construction in this class, where the only things change is the code at the top of the pyramid (in this case L38). Extract a common method

```kt
doWithCredentials { dispatcher.dispatch(RouteAction.OpenWebsite(it.formSubmitURL!!)) }
```